src/Cms/AppPlugins.php

```diff
@@ -871,18 +872,35 @@ public static function plugin(
 		string|null $root = null,
 		string|null $version = null,
 		Closure|string|array|null $license = null,
+		bool|string $autoloader = false
 	): Plugin|null {
-		if ($extends === null) {
+
+
+		if ($extends === null && $autoloader === false) {
 			return static::$plugins[$name] ?? null;
 		}
 
+		$root ??= $extends['root'] ?? dirname(debug_backtrace()[0]['file']);
+
+		if ($autoloader) {
+
+			//Allow to apply custom Autoloader
+			$autolader_class = is_bool($autoloader) ? Autoloader::class : $autoloader;
+
+			$extends = $autolader_class::load(
+				name: $name,
+				root: $root,
+				data: $extends ?? []
```

Comment:
`Suggestion:` Instead of passing `$extends` to the class and merging it in the class's `::toArray()` method, the class could be responsible to only gather and return autoloader extensions. And then we rather merge the explicitly passed `$extends` and the result of `Autoloader::toArray()` here.